controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -244,6 +271,19 @@ func (r *ReconcileHyperConverged) Reconcile(ctx context.Context, request reconci
 	if hcoTriggered {
 		logger.Info("Reconciling HyperConverged operator")
 		r.operandHandler.Reset()
+
+		apiServerCRTriggered, err := isTriggeredByApiServerCR(request)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if apiServerCRTriggered {
+			logger.Info("Triggered by ApiServer CR, refreshing it")
+			err = hcoutil.GetClusterInfo().RefreshAPIServerCR(hcoRequest.Ctx, r.client)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+
```

Comment:
SonarCloud warns about a code smell here (too complex function). Please consider to refactor.

Reply:
done